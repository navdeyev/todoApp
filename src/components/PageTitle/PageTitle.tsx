import * as React from 'react';

import {Heading1} from 'components/Styled/Styled';
import styled from 'components/Styled/styledComponents';
import {defaultTheme, IHasTheme} from 'components/Styled/themes';

const Title: React.SFC<IHasTheme> = props =>
    <Heading1 className={props.className}>{props.children}</Heading1>;

const PageTitle = styled(Title)`color: ${ props => props.theme.pageTitleColor };`;

PageTitle.defaultProps = {theme: defaultTheme};
PageTitle.displayName = 'PageTitle';

export default PageTitle;
