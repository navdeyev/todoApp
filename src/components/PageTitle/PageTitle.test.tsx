import {shallow} from 'enzyme';
import * as React from 'react';

import PageTitle, {IPageTitleProps} from './PageTitle';

describe('PageTitle', () => {
    it('renders a themed PageTitle component', () => {
        const props: IPageTitleProps = {
            children: 'Some text here'
        };
        const render = shallow(<PageTitle {...props}/>);
        expect(render.getElement()).toMatchSnapshot();
    });
});
