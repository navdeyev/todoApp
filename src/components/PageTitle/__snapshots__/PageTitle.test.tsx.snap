// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders a themed PageTitle component 1`] = `
<Title
  className="sc-bxivhb eARnsh"
  theme={
    Object {
      "hoverStatusBorderColor": "#222222",
      "pageBackgroundColor": "#ffffff",
      "pageHeaderBgColor": "#222222",
      "pageMainTextColor": "#000000",
      "pageTitleColor": "#ffffff",
      "statusBorderColor": "#666666",
      "todoCardBorderColor": "#666666",
    }
  }
>
  Some text here
</Title>
`;
